src/components/Sweep/SweepApprove.tsx

```diff
@@ -0,0 +1,141 @@
+import { useCallback, useState, useEffect } from 'react';
+import { Form, Input, Button, notification, Typography } from 'antd';
+import { ethers } from 'ethers';
+import styled from 'styled-components';
+
+import { numberToBn, ethereumAddressValidator, contractApprove, contractBalanceOf } from '../../utils';
+
+import { useMetamask, requestAccount } from '../../hooks/useMetamask';
+
+function SweepApprove() {
+  const [loading, setLoading] = useState(false);
+  const [valid, setValid] = useState(false);
+
+  const [spenderAddr, setSpenderAddr] = useState('');
+  const [approveAmount, setApproveAmount] = useState(0);
+  const [maxAmount, setmaxAmount] = useState(0);
+
+  const { changeNetwork } = useMetamask();
+
+  const handleSubmit = useCallback(async () => {
+    const validAddress = ethers.utils.isAddress(spenderAddr);
+    if (validAddress) {
+      try {
+        setLoading(true);
+        await changeNetwork(42161);
+        const account = await requestAccount();
+
+        if (approveAmount.toString() === '') {
+          await contractApprove(spenderAddr, account, ethers.constants.MaxUint256);
+        } else {
+          await contractApprove(spenderAddr, account, numberToBn(approveAmount));
+        }
+
+        notification.success({
+          message: `Wallet ${spenderAddr} has been whitelisted for ${approveAmount} amount succesffuly. `,
+        });
+      } catch (e) {
+        console.error(e);
+        const error = (e as any)?.message || 'Error';
+        notification.error({ message: error });
+      } finally {
+        setLoading(false);
+      }
+    }
+  }, [spenderAddr, approveAmount, changeNetwork]);
+
+  const handleSpenderAddrChange = useCallback(
+    (e) => {
+      setSpenderAddr(e.target.value);
+    },
+    [setSpenderAddr]
+  );
+
+  const handleAmountChange = useCallback(
+    (e) => {
+      setApproveAmount(e.target.value);
+    },
+    [setApproveAmount]
+  );
+
+  const setAmountToMax = () => setApproveAmount(maxAmount);
+
+  useEffect(() => {
+    const getBalance = async () => {
+      const account = await requestAccount();
+      const validAFromAddress = ethers.utils.isAddress(account);
```

Comment:
what is `A` for in the name? also, the proper name for this boolean value is `isValidFromAddress`. `validFromAddress` implies a value that holds an address (string or similar) that is valid, not boolean